### ISSUE TITLE
content_tag_override in yaml does not appear to be working

### ISSUE THREAD

**Issue by blalbrit** [sul-dlss/pre-assembly#36]
with flag set to true, project style content structure ignores registered content type. with flag set to false, project style content structure also ignores registered content type.

would expect one of those options to allow registered content tags to determine pre-assembly content type.

**Comment by atz** [comment]
@blalbrit Need more specificity about what class is being used, `Bundle` or `DigitalObject`?  In both cases, the classes apply the incoming params, **then** initialize the default values on top of them (i.e. ignoring those params).  The design is questionable.

**Comment by peetucket** [comment]
I think for the context of this short workcycle, we decided this parameter would be constant therefore perhaps making this bug out of scope (unless there is time?).

My recollection of the way these incoming YAML parameters work is that the `Bundle` object gets set with everything and then passes the values to each instance of `DigitalObject` .  This is not a great design (since the values are the same for each digitalobject) but wouldn't explain the problem behavior.

**Comment by blalbrit** [comment]
Agreed re: scope. However, if we're going to lock it to "true" in the form and make that parameter constant, we do need to verify that it will obey the registered tags.

**Comment by peetucket** [comment]
Got it.  Looks like there are some tests around this behavior of this parameter:

https://github.com/sul-dlss/pre-assembly/blob/master/spec/digital_object_spec.rb#L461-L503
https://github.com/sul-dlss/pre-assembly/blob/master/spec/digital_object_spec.rb#L591-L601

**Comment by peetucket** [comment]
Hmmm...I wonder if the problem is coming from here:
https://github.com/sul-dlss/pre-assembly/blob/master/lib/pre_assembly/digital_object.rb#L94

Perhaps instead of an "or" in that first part of the conditional, it should be an "and"  : `(@project_style[:should_register]) || (!@project_style[:content_tag_override])` 
should be
 `(@project_style[:should_register]) && (!@project_style[:content_tag_override])`

That block of code is confusing but I'm not sure I'm grokking the logic right now.

On the other hand, the tests do check for the value of `content_md_creation_style` which is what comes out of that method.  Maybe the tests are missing a scenario or are not correctly validating the value that is set.

**Comment by peetucket** [comment]
Actually I think the logic did make sense; but was confusing ... started a branch to try to make that logic a bit easier to understand....tests still pass.

@blalbrit Maybe the next step is find an example of a YAML file with some objects where the behavior was *not* working as expected?  Maybe there is something simple like an upper/lowercase issue in the content type coming from object?

**Comment by peetucket** [comment]
Current tests appear to cover the override scenario and are working as expected.
